sormas-app/app/src/main/java/de/symeda/sormas/app/immunization/edit/ImmunizationEditActivity.java

```diff
@@ -0,0 +1,164 @@
+
+import java.util.List;
+
+import de.symeda.sormas.api.utils.DataHelper;
+import de.symeda.sormas.api.utils.ValidationException;
+import de.symeda.sormas.app.BaseActivity;
+import de.symeda.sormas.app.BaseEditActivity;
+import de.symeda.sormas.app.BaseEditFragment;
+import de.symeda.sormas.app.R;
+import de.symeda.sormas.app.backend.common.DaoException;
+import de.symeda.sormas.app.backend.common.DatabaseHelper;
+import de.symeda.sormas.app.backend.immunization.Immunization;
+import de.symeda.sormas.app.backend.immunization.ImmunizationEditAuthorization;
+import de.symeda.sormas.app.component.menu.PageMenuItem;
+import de.symeda.sormas.app.component.validation.FragmentValidator;
+import de.symeda.sormas.app.core.async.AsyncTaskResult;
+import de.symeda.sormas.app.core.async.SavingAsyncTask;
+import de.symeda.sormas.app.core.async.TaskResultHolder;
+import de.symeda.sormas.app.core.notification.NotificationHelper;
+import de.symeda.sormas.app.immunization.ImmunizationSection;
+import de.symeda.sormas.app.person.edit.PersonEditFragment;
+import de.symeda.sormas.app.util.Bundler;
+
+import static de.symeda.sormas.app.core.notification.NotificationType.ERROR;
+import static de.symeda.sormas.app.core.notification.NotificationType.WARNING;
+
+public class ImmunizationEditActivity extends BaseEditActivity<Immunization> {
+
+	private AsyncTask saveTask;
+
+	public static void startActivity(Context context, String rootUuid) {
+		BaseEditActivity.startActivity(context, ImmunizationEditActivity.class, buildBundle(rootUuid));
+	}
+
+	public static void startActivity(Context context, String recordUuid, ImmunizationSection section) {
+		BaseActivity.startActivity(context, ImmunizationEditActivity.class, buildBundle(recordUuid, section));
+	}
+
+	public static Bundler buildBundle(String recordUuid, ImmunizationSection section) {
+		return BaseEditActivity.buildBundle(recordUuid, section.ordinal());
+	}
+
+	@Override
+	protected Immunization queryRootEntity(String recordUuid) {
+		return DatabaseHelper.getImmunizationDao().queryUuidWithEmbedded(recordUuid);
+	}
+
+	@Override
+	protected Immunization buildRootEntity() {
+		throw new UnsupportedOperationException();
+	}
+
+	@Override
+	public List<PageMenuItem> getPageMenuData() {
+		return PageMenuItem.fromEnum(ImmunizationSection.values(), getContext());
+	}
+
+	@Override
+	protected BaseEditFragment buildEditFragment(PageMenuItem menuItem, Immunization activityRootData) {
+		ImmunizationSection section = ImmunizationSection.fromOrdinal(menuItem.getPosition());
+		BaseEditFragment fragment;
+		switch (section) {
+
+		case PERSON_INFO:
+			fragment = PersonEditFragment.newInstance(activityRootData);
+			break;
+		case IMMUNIZATION_INFO:
+			fragment = ImmunizationEditFragment.newInstance(activityRootData);
+			break;
+		default:
+			throw new IndexOutOfBoundsException(DataHelper.toStringNullable(section));
+		}
+
+		return fragment;
+	}
+
+	@Override
+	public void saveData() {
+		if (saveTask != null) {
```

Comment:
If I'm not mistaken and not missing anything, duplicate detection is missing here; please have another look at the issue description; there's supposed to be a popup if the user changes the start and end date and then tries to save when there's an existing overlapping immunization.